src/Stryker.Core/Stryker.Core/Initialisation/FsharpProjectComponentsBuilder.cs

```diff
@@ -0,0 +1,241 @@
+using Stryker.Core.ProjectComponents;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using static FSharp.Compiler.SyntaxTree.ParsedInput;
+using ParsedInput = FSharp.Compiler.SyntaxTree.ParsedInput;
+
+namespace Stryker.Core.Initialisation
+{
+    internal class FsharpProjectComponentsBuilder : ProjectComponentsBuilder
+    {
+        private readonly ProjectInfo _projectInfo;
+        private readonly string[] _foldersToExclude;
+        private readonly ILogger _logger;
+        private readonly IFileSystem _fileSystem;
+
+        public FsharpProjectComponentsBuilder(ProjectInfo projectInfo, string[] foldersToExclude, ILogger logger, IFileSystem fileSystem)
+        {
+            _projectInfo = projectInfo;
+            _foldersToExclude = foldersToExclude;
+            _logger = logger;
+            _fileSystem = fileSystem;
+        }
+        
+        public override IProjectComponent Build()
+        {
+            FsharpFolderComposite inputFiles;
+            if (_projectInfo.ProjectUnderTestAnalyzerResult.SourceFiles != null && _projectInfo.ProjectUnderTestAnalyzerResult.SourceFiles.Any())
+            {
+                inputFiles = FindProjectFilesUsingBuildalyzer(_projectInfo.ProjectUnderTestAnalyzerResult);
+            }
+            else
+            {
+                inputFiles = FindProjectFilesScanningProjectFolders(_projectInfo.ProjectUnderTestAnalyzerResult);
+            }
+            return inputFiles;
+        }
+        private FsharpFolderComposite FindProjectFilesUsingBuildalyzer(IAnalyzerResult analyzerResult)
+        {
+            var inputFiles = new FsharpFolderComposite();
+            var projectUnderTestDir = Path.GetDirectoryName(analyzerResult.ProjectFilePath);
+            var projectRoot = Path.GetDirectoryName(projectUnderTestDir);
+            var rootFolderComposite = new FsharpFolderComposite()
+            {
+                FullPath = projectRoot,
+                RelativePath = string.Empty
+            };
+            var cache = new Dictionary<string, FsharpFolderComposite> { [string.Empty] = rootFolderComposite };
+
+            // Save cache in a singleton so we can use it in other parts of the project
+            FolderCompositeCache<FsharpFolderComposite>.Instance.Cache = cache;
+
+            inputFiles.Add(rootFolderComposite);
+
+            var fSharpChecker = FSharpChecker.Create(projectCacheSize: null, keepAssemblyContents: null, keepAllBackgroundResolutions: null, legacyReferenceResolver: null, tryGetMetadataSnapshot: null, suggestNamesForErrors: null, keepAllBackgroundSymbolUses: null, enableBackgroundItemKeyStoreAndSemanticClassification: null);
+
+            foreach (var sourceFile in analyzerResult.SourceFiles)
+            {
+                // Skip xamarin UI generated files
+                if (sourceFile.EndsWith(".xaml.cs"))
+                {
+                    continue;
+                }
```

Comment:
Are other generated files (Like for resx files) also accounted for?

Reply:
resx files need to be included, they cannot be skipped